frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx

```diff
@@ -1,22 +1,34 @@
 import type { EmotionCache } from "@emotion/react";
 import { withEmotionCache } from "@emotion/react";
+import type { MantineThemeOverride } from "@mantine/core";
 import { MantineProvider } from "@mantine/core";
-import type { ReactNode } from "react";
+import { merge } from "icepick";
+import { useMemo, type ReactNode } from "react";
 
 import { getThemeOverrides } from "../../../theme";
 import { DatesProvider } from "../DatesProvider";
 
 interface ThemeProviderProps {
   children: ReactNode;
+
+  /**
+   * Extend Metabase's theme overrides.
+   * This is primarily used in the React embedding SDK
+   * to allow SDK users to customize the theme.
+   */
+  theme?: MantineThemeOverride;
```

Comment:
should we use `MetabaseTheme` like we pass it in `enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx` for consistency?

Reply:
Fair enough. Also, I thought there's a way to type `theme.other`, but it doesn't look like Mantine allow that either.

Reply:
@WiNloSt in the Mantine V6 docs, they mentioned we are supposed to use declaration merging to define this. It's not in the Mantine V7 docs, but I think this method still works as the type is still `MantineThemeOther`.

![CleanShot 2567-05-02 at 16 01 24@2x](https://github.com/metabase/metabase/assets/4714175/2771af38-853c-4a96-aed6-7d2bb1a77ab8)